src/components/Atoms/Button/Button.tsx

```diff
@@ -27,18 +36,45 @@ export const ButtonBase = styled('button').withConfig({
 export type ButtonProps = Omit<JSX.IntrinsicElements['button'], 'type'> & TButton;
 export type ButtonComponent = PolymorphicComponent<ButtonProps>;
 
+// eslint-disable-next-line react/function-component-definition
 export const Button: ButtonComponent = (props) => {
-	const { children, buttonColor, ...buttonProps } = props;
+	const {
+		children, trailingIcon, leadingIcon, buttonColor = 'primary', variant = 'primary', size = 'large', textAlign = 'center', ...buttonProps
+	} = props;
 	const theme = useTheme();
 	const color = (!buttonColor && theme.buttonColor[buttonProps.variant]) ? buttonProps.variant : buttonColor;
-	return <ButtonBase {...buttonProps} buttonColor={color}>{children}</ButtonBase>;
-};
+	const isTextLink = variant === 'textLink';
 
-Button.defaultProps = {
-	buttonColor: undefined,
-	initialDisplay: 'inline-block',
-	size: 'large',
-	textAlign: 'center',
-	variant: 'primary',
-	whiteSpace: 'nowrap',
+	return (
+		<ButtonBase
+			{...{
+				size, variant, textAlign, ...buttonProps,
+			}} buttonColor={color}
+		>
+			<Flex gap={isTextLink ? '6px' : '4px'} alignItems="center" h="100%">
+				{leadingIcon && !isTextLink && (
+					<Flex>
+						<Icon name={leadingIcon} size={calculateIconSize(buttonProps.size)} color={JSON.stringify(color)} />
+					</Flex>
+				)}
+				<Wrapper paddingX={calculatePaddingX(variant)} paddingBottom={calculatePaddingBottom(variant)}>
+					{children}
+				</Wrapper>
+				{trailingIcon && (
+					isTextLink ? (
+						<Flex w="24px" flexShrink="0">
+							<Icon
+								id="text-link-trailing-icon" name={trailingIcon} size="20"
```

Comment:
@Ivana-Jovic zasto je ovde potreban id?

Reply:
@AleksaStevic Isti razlog kao gore

Reply:
@Ivana-Jovic Jel mozes da probas preko varijanti to da rešiš? mnogo mi je ruzno ovde da to sve stoji jer je prevelika komplikacija

Reply:
@AleksaStevic Ako mozes da bacis pogled kako je sad, i da vidis da li je bolje. I ako bi i dalje da prebacimo u varijante, ako mozes da mi pojasnis kako  tacno mislis